LibreNMS/Data/Store/Rrd.php

```diff
@@ -481,25 +483,26 @@ public function purge($hostname, $prefix)
 
     /**
      * Generates a graph file at $graph_file using $options
-     * Opens its own rrdtool pipe.
+     * Graphs are a single command per run, so this just runs rrdtool
      *
-     * @param string $graph_file
-     * @param string $options
-     * @return string|int
+     * @param  string  $options
+     * @return string
+     * @throws \LibreNMS\Exceptions\FileExistsException
+     * @throws \LibreNMS\Exceptions\RrdGraphException
      */
-    public function graph($graph_file, $options)
+    public function graph(string $options): string
     {
-        if ($this->init(false)) {
-            $cmd = $this->buildCommand('graph', $graph_file, $options);
-
-            $output = implode($this->sync_process->sendCommand($cmd));
-
-            d_echo("<p>$cmd</p>\n<p>command returned ($output)</p>");
-
-            return $output;
-        } else {
-            return 0;
+        $cmd = $this->buildCommand('graph', '-', $options);
+        $process = Process::fromShellCommandline(Config::get('rrdtool') . ' ' . $cmd);
```

Comment:
Since we're not doing any redirection or so, why aren't we building an array as argument, and use the "normal" command? So we don't have to worry about escaping and such?
Or can `$options` contain that? 

> If you need to use stream redirections, conditional execution, or any other feature provided by the shell of your operating system, you can also define commands as strings using the fromShellCommandline() static factory.
> 
> Each operating system provides a different syntax for their command-lines, so it becomes your responsibility to deal with escaping and portability.

Reply:
Because I don't want to change 20,000 lines of code to fix every single graph.  The way graphs were designed in Observium is to concatenate rrdtool options onto a string and run that.